resources/splunk_app.rb

```diff
@@ -69,25 +65,43 @@ def parse_meta_access(access)
     end
   end
 
+  # Provider exclusive methods
+
   action_class do
     include CernerSplunk::ProviderHelpers
 
+    def upgrade_app(strategy)
```

Comment:
I have some desire to contain these methods under one umbrella, for easy identification most of all. I think I can make these methods such as `backup` and `keep_existing` under a module `AppUpgrade`. Thoughts?